src/effects/backends/lv2/lv2manifest.cpp

```diff
@@ -60,29 +66,58 @@ LV2Manifest::LV2Manifest(const LilvPlugin* plug,
 
             // Get and set the parameter name
             info = lilv_port_get_name(m_pLV2plugin, port);
-            QString paramName = lilv_node_as_string(info);
-            param->setName(paramName);
+            param->setName(lilv_node_as_string(info));
             lilv_node_free(info);
 
             const LilvNode* node = lilv_port_get_symbol(m_pLV2plugin, port);
-            QString symbol = lilv_node_as_string(node);
-            param->setId(symbol);
+            param->setId(lilv_node_as_string(node));
             // node must not be freed here, it is owned by port
 
-            param->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+            EffectManifestParameter::UnitsHint unitsHint =
+                    EffectManifestParameter::UnitsHint::Unknown;
+            LilvNodes* units = lilv_port_get_value(m_pLV2plugin, port, properties["unit"]);
+            if (lilv_nodes_size(units) > 0) {
+                LilvNode* unit = lilv_nodes_get_first(units);
+                // If this is a unit symbol URI, e.g. http://lv2plug.in/ns/extensions/units#ms
+                if (lilv_node_is_uri(unit)) {
+                    QString unitStr = lilv_node_as_uri(unit);
+                    // that starts with the 'units' prefix, isolate the identifier
+                    if (unitStr.startsWith(lilv_node_as_string(properties["unit_prefix"]))) {
+                        unitsHint = EffectManifestParameter::lv2UnitToUnitsHint(
+                                unitStr.remove(lilv_node_as_string(properties["unit_prefix"])));
+                    }
+                } else { // Try to extract the custom unit symbol string
+                    LilvNode* customUnit =
+                            lilv_world_get(world, unit, properties["unit_symbol"], NULL);
+                    if (customUnit) {
+                        // Accepted custom units needs to be 'whitelisted' in
+                        // EffectManifestParameter::lv2UnitToUnitsHint and added to
+                        // EffectManifestParameter::unitsHintToString
+                        unitsHint = EffectManifestParameter::lv2UnitToUnitsHint(
+                                lilv_node_as_string(customUnit));
+                        if (lv2ParamDebug &&
```

Comment:
well, yes, but in my case with the LSP suite it prints hundreds of lines with `G`

Reply:
we can leave it enabled until the release so we can easily look for missing units?

Reply:
Since we know of the G case and have decided to use no unit for it. How about skip the message in this case. This also gives a chance to drop a comment about to our decision.

Reply:
okay, for `G` it's a simple check. if the blacklist grows we need to rework this though.